src/InfoViz/Native/HistogramSelector/index.js

```diff
@@ -770,33 +721,58 @@ function histogramSelector(publicAPI, model) {
     });
   }
 
+  // Auto unmount on destroy
   model.subscriptions.push({ unsubscribe: publicAPI.setContainer });
-  // event from the FieldProvider
-  // TODO overkill if one field's 'active' flag changes.
-  model.subscriptions.push(model.provider.onFieldChange(fetchData));
-  // event from Histogram Provider
-  model.subscriptions.push(model.provider.onHistogram1DReady(publicAPI.render));
+
+  if (model.provider.isA('FieldProvider')) {
+    const fieldNames = model.provider.getFieldNames();
+    if (!model.fieldData) {
+      model.fieldData = {};
+    }
+
+    fieldNames.forEach(name => {
+      model.fieldData[name] = Object.assign(
+        model.fieldData[name] || {},
+        model.provider.getField(name),
+        scoreHelper.defaultFieldData());
+    });
+
+    model.subscriptions.push(model.provider.onFieldChange(field => {
+      Object.assign(model.fieldData[field.name], field);
+    }));
+  }
 
   if (model.provider.isA('HistogramBinHoverProvider')) {
     model.subscriptions.push(model.provider.onHoverBinChange(handleHoverUpdate));
   }
 
-  if (model.provider.isA('DataUpdateProvider')) {
-    model.updateSubscription = model.provider.subscribeToDataUpdate(
-      'histogram1d',
+  if (model.provider.isA('Histogram1DProvider')) {
+    model.histogram1DDataSubscription = model.provider.subscribeToHistogram1D(
       data => {
+        if (!model.histograms) {
+          model.histograms = {};
```

Comment:
We've lost data sharing - FieldSelector, HistogramSelector and MI all have their own copies of the 1D histogram.